src/Impute.jl

```diff
@@ -1,14 +1,40 @@
 module Impute
 
-using DataFrames
+using IterTools
 using Statistics
+using StatsBase
+using Tables: Tables, materializer, istable
 
-import DataFrames: DataFrameRow
-import Base.Iterators
+import Base.Iterators: drop
 
 export impute, impute!, chain, chain!, drop, drop!, interp, interp!, ImputeError
 
-const Dataset = Union{AbstractArray, DataFrame}
+function __init__()
+    sym = join(["chain", "chain!", "drop", "drop!", "interp", "interp!"], ", ", " and ")
+
+    @warn(
+        """
+        The following symbols will not be exported in future releases: $sym.
+        Please qualify your calls with `Impute.<method>(...)` or explicitly import the symbol.
+        """
+    )
+
+    @warn(
+        """
+        The default limit for all impute functions will be 1.0 going forward.
+        If you depend on a specific threshold please pass in an appropriate `AbstractContext`.
+        """
+    )
+
+    @warn(
+        """
+        All matrix imputation methods will be switching to the column-major convention
+        (e.g., each column corresponds to an observation, and each row corresponds to a variable).
+        To maintain the existing behaviour please pass `vardim=2` to the `Imputor` constructors
```

Comment:
Should `vardim` belong to the `Context`?

Reply:
I don't think so because the `Imputor` is responsible for iterating through the datastructure and not the `Context`.